qiskit_nature/circuit/library/ansatzes/succd.py

```diff
@@ -106,27 +106,43 @@ def include_singles(self, include_singles: Tuple[bool, bool]) -> None:
         """Sets whether to include single excitations."""
         self._include_singles = include_singles
 
+    @property
+    def full(self) -> bool:
+        """Whether use SUCC_full"""
+        return self._full
+
+    @full.setter
+    def full(self, full: bool) -> None:
+        """Sets whether to use SUCC_full."""
+        self._full = full
+
+    @property
+    def excitation_list(self) -> List[Tuple[Tuple[int, ...], Tuple[int, ...]]]:
+        """The excitation list that SUCC is using, in the required format."""
+        return self.generate_excitations(
+            self.num_spin_orbitals,
+            self.num_particles,
+        )
```

Comment:
There should be no need for this code. `self.generate_excitations` is already injected as the excitation generator during `__init__`.

Reply:
I kept this part, because if I remove it, then "**self.excitation_list**" will return SUCCD excitation list even we set "**full=True**"

Reply:
This is actually not true. Although it looks surprisingly similar (simply because the length matches what you expected).
However, you are running into a different issue here. During the operator conversion from second-quantization to qubit space, the UCC class filters the operators based on their commutation properties: https://github.com/Qiskit/qiskit-nature/blob/b0be0fbe4ece34a643339a1d838a0a86e752a8b1/qiskit_nature/second_q/circuit/library/ansatzes/ucc.py#L287-L300

More specifically, the `zip(...)` operation on this line: https://github.com/Qiskit/qiskit-nature/blob/b0be0fbe4ece34a643339a1d838a0a86e752a8b1/qiskit_nature/second_q/circuit/library/ansatzes/ucc.py#L297
is what is causing issues for you, because your `operators` has length 3, whereas `self._excitation_list` has length 4 (for the specific test case in your unittest). This results in the last excitation being dropped.

I don't have a perfect solution right now, but I believe what needs to be done here, is that this validation logic needs to be extracted into a private method. `SUCCD` can then overwrite this method and use the `excutations_dictionary` instead of `self._excitation_list`, since the former should have the correct length again.

Reply:
If I keep the code showing here, that means I override **self._excitation_list**  in SUCCD, so it doesn't have to have the same length as **operators**. In this way, **self.excitation_list** will return the correct list for both SUCCD and SUCC_full. Do you think this is a feasible solution?

Reply:
I would think the better approach would be, that you slightly refactor the `UCC` class and extract that validation logic into a separate private method.
In `SUCCD` you can then overwrite this validation logic, to use your internal excitation dictionary instead of the default validation logic